src/libstd/path.rs

```diff
@@ -999,16 +1036,58 @@ impl PathBuf {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<P: AsPath> iter::FromIterator<P> for PathBuf {
+impl<'a> From<&'a str> for PathBuf {
```

Comment:
Yes, the inability to do blanket "transportation" of one conversion to another is a loss here.

Note that specialization may someday allow us to regain this via automatic reflexive and transitive blankets. It's not clear.